src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs

```diff
@@ -21,6 +21,9 @@ namespace Microsoft.VisualStudio.IntegrationTest.Utilities
 {
     public sealed class VisualStudioInstanceFactory : IDisposable
     {
+        [ThreadStatic]
+        private static bool s_inHandler;
```

Comment:
You _have_ to use `ThreadLocal<T>` if you need to supply an initializer. `ThreadLocal<T>` also needs to be disposed. That wouldn't be a problem here, but `ThreadStatic` is sufficient for our purposes.